lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -1096,9 +1098,11 @@ instance ToJSON ApiByronWalletBalance where
     toJSON = genericToJSON defaultRecordTypeOptions
 
 instance FromJSON (ApiT PoolId) where
-    parseJSON = parseJSON >=> eitherToParser . bimap ShowFmt ApiT . fromText
+    parseJSON = parseJSON >=> eitherToParser
+           . bimap ShowFmt ApiT
+           . decodePoolIdBech32
```

Comment:
@hasufell I'm happy to continue discussing editor configs in our slack channel, if you feel it would help.

Though for this PR, there is only this request:

```suggestion
    parseJSON = parseJSON >=> eitherToParser
       . bimap ShowFmt ApiT
       . decodePoolIdBech32
```
and
https://github.com/input-output-hk/cardano-wallet/pull/2055/files#r477206873

If you can wrap these up, then we can merge this PR immediately. :+1: